src/main/java/pro/sky/JavaCoredemo2/services/JavaQuestionService.java

```diff
@@ -0,0 +1,58 @@
+package pro.sky.JavaCoredemo2.services;
+
+import pro.sky.JavaCoredemo2.entity.Question;
+import pro.sky.JavaCoredemo2.exceptions.NoQuestionsException;
+import pro.sky.JavaCoredemo2.repository.JavaQuestionRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+public class JavaQuestionService implements QuestionService {
+
+    private final Random random = new Random();
+    private final JavaQuestionRepository repository;
+
+    public JavaQuestionService(JavaQuestionRepository repository) {
```

Comment:
```suggestion
    public JavaQuestionService(QuestionRepository repository) {
```
инъекцию надо всегда делать по интерфейсу, а не реализации

Reply:
также см. внимательнее задание про Qualifier аннотацию